### PR TITLE
Thread changes

### PR DESCRIPTION
**Pull request by AnttiKauppila**
@mikter Review & merge

### CODE REVIEW

File: coap-service/coap_service_api.h [hunk 1]
```diff
@@ -106,38 +86,58 @@ typedef int coap_service_security_start_cb(int8_t service_id, uint8_t EUI64[stat
...
+ * \brief Close secure connection
  *
- * \return 0 for success / -1 for failure 
-  */
-typedef int coap_service_response_recv(int8_t service_id, uint16_t msg_id, sn_coap_hdr_s *response_ptr);
+ * Closes secure connection (if present), but leaves socket open.
+ *
+ * \param service_id         Id number of the current service.
+ */
+extern void coap_service_close_secure_connection(int8_t service_id);
```
Comment (mikter) [inline coap-service/coap_service_api.h]: this needs remote address and port to close one specific session one service can have many sessions and might want to close only one

File: coap-service/coap_service_api.h [hunk 2]
```diff
@@ -204,41 +204,9 @@ int8_t coap_service_register_uri(int8_t service_id, const char *uri, uint8_t all
...
- *
- * \param service_id        Id number of the current service.
- * \param *start_ptr        Callback to inform security handling is started.
- * \param *security_done_cb Callback to inform security handling is done.
- *
- * \return -1               for failure 
- *          instance_id     For success (is used to identify registery)
- *
- */
-int8_t coap_service_register_uri_secure_cb_set(int8_t service_id, coap_service_security_start_cb *start_ptr, coap_service_security_done_cb *security_done_cb);
```
Comment (mikter) [inline coap-service/coap_service_api.h]: this functionshould be back user might want to change the callbacks even after service is created and now it is impossible

File: coap-service/coap_service_api.h [hunk 3]
```diff
@@ -106,38 +86,58 @@ typedef int coap_service_security_start_cb(int8_t service_id, uint8_t EUI64[stat
...
+ * \param listen_port        Port that Application wants to use for communicate with coap server.
+ * \param service_options    Options of the current service.
+ * \param *start_ptr         Callback to inform security handling is started and to fetch device password.
+ * \param *security_done_cb  Callback to inform security handling is done.
  *
- * \return 0 for success / -1 for failure 
+ *  \return service_id / -1 for failure
  */
-int coap_service_security_key_set(int8_t service_id, uint8_t EUI64[static 8], uint8_t *PSKd_ptr, uint8_t PSKd_len);
+extern int8_t coap_service_initialize(int8_t interface_id, uint16_t listen_port, uint8_t service_options, coap_service_security_start_cb *start_ptr, coap_service_security_done_cb *security_done_cb);
```
Comment (mikter) [inline coap-service/coap_service_api.h]: I would not want to enable callbacks before I am certain that I want someone to connect so these should not be in initialisation function, but in the own setter function

File: coap-service/coap_service_api.h (hunk 2 above)
Comment (AnttiKauppila) [inline coap-service/coap_service_api.h]: This was never used as you say

File: coap-service/coap_service_api.h (hunk 3 above)
Comment (AnttiKauppila) [inline coap-service/coap_service_api.h]: And still you did in your old code?

**Comment by AnttiKauppila** [comment]
rebuild

**Pull request merged**